src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -94,16 +189,41 @@ public void controlProperties(List<ControllableProperty> controllableProperties)
 	 * @return List<Statistics> This return the list of statistics.
 	 */
 	@Override
-	public List<Statistics> getMultipleStatistics() {
-		Map<String, String> stats = new HashMap<>();
-		failedMonitor = new HashMap<>();
-		loginInfo = initLoginInfo();
+	public List<Statistics> getMultipleStatistics() throws Exception {
 		ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+		// This is to make sure if the statistics is being fetched before/after any set of control operations
+		reentrantLock.lock();
+		try {
+			if (qrcCommunicator == null) {
+				initQRCCommunicator();
+			}
+
+			if (isEmergencyDelivery && localExtStat != null) {
+				isEmergencyDelivery = false;
+			} else {
+				Map<String, String> stats = new HashMap<>();
+				List<AdvancedControllableProperty> controllableProperties = new ArrayList<>();
 
-		populateQSYSMonitoringMetrics(stats);
-		extendedStatistics.setStatistics(stats);
+				failedMonitor = new HashMap<>();
+				loginInfo = initLoginInfo();
+
+				populateQSYSMonitoringMetrics(stats);
+				populateGainControllingMetrics(stats, controllableProperties);
+
+				extendedStatistics.setStatistics(stats);
+				extendedStatistics.setControllableProperties(controllableProperties);
+				localExtStat = extendedStatistics;
+			}
 
-		return Collections.singletonList(extendedStatistics);
+			if (updateLocalExtStatDto != null) {
+				updateLocalExtStat(updateLocalExtStatDto);
+				updateLocalExtStatDto = null;
+			}
+		} finally {
+			reentrantLock.unlock();
+		}
+
+		return Collections.singletonList(localExtStat);
 	}
 
 	/**
```

Comment:
@inheritDoc here since it overrides the javadoc from superclass. It should also inherit the doc

Reply:
thank you, I will fix it